---
title: "es-x/no-accessor-properties"
description: "disallow accessor properties"
since: "[eslint-plugin-es] v1.1.0"
---

# es-x/no-accessor-properties
> disallow accessor properties

- âœ… The following configurations enable this rule: [no-new-in-es5] and [restrict-to-es3]

This rule reports ES5 accessor properties as errors.

## ðŸ’¡ Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad">

```js
/*eslint es-x/no-accessor-properties: error */
var a = {
    get a() {},
    set a(value) {}
}
class A {
    get a() {}
    set a(value) {}
}
```

</eslint-playground>

## ðŸš€ Version

This rule was introduced in [eslint-plugin-es] v1.1.0.

[eslint-plugin-es]: https://github.com/mysticatea/eslint-plugin-es

## ðŸ“š References

- [Rule source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/lib/rules/no-accessor-properties.js)
- [Test source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/tests/lib/rules/no-accessor-properties.js)

[no-new-in-es5]: ../configs/index.md#no-new-in-es5
[restrict-to-es3]: ../configs/index.md#restrict-to-es3
