# es-x/no-block-scoped-functions
> disallow block-scoped function declarations

- âœ… The following configurations enable this rule: `plugin:es-x/no-new-in-es2015`, `plugin:es-x/restrict-to-es3`, and `plugin:es-x/restrict-to-es5`

This rule reports ES2015 block-scoped function declarations as errors.

## Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad" code="/*eslint es-x/no-block-scoped-functions: error */
if (a) {
    function f() {}
} else {
    function g() {}
}
" />

## ðŸ“š References

- [Rule source](https://github.com/ota-meshi/eslint-plugin-es-x/blob/v4.1.0/lib/rules/no-block-scoped-functions.js)
- [Test source](https://github.com/ota-meshi/eslint-plugin-es-x/blob/v4.1.0/tests/lib/rules/no-block-scoped-functions.js)
