# disallow `for-of` statements (es/no-for-of-loops)

This rule reports ES2015 `for-of` statements as errors.

## Examples

â›” Examples of **incorrect** code for this rule:

```js
for (var a of b) {}
for (let a of b) {}
for (a of b) {}
```

## ðŸ“š References

- [Rule source](../../lib/rules/no-for-of-loops.js)
- [Test source](../../tests/lib/rules/no-for-of-loops.js)
