---
title: "es-x/no-function-prototype-bind"
description: "disallow the `Function.prototype.bind` method"
since: "v5.0.0"
---

# es-x/no-function-prototype-bind
> disallow the `Function.prototype.bind` method

- âœ… The following configurations enable this rule: `plugin:es-x/no-new-in-es5` and `plugin:es-x/restrict-to-es3`

This rule reports ES5 `Function.prototype.bind` method as errors.

This rule is silent by default because it's hard to know types. You need to configure [the aggressive mode](../#the-aggressive-mode) or TypeScript in order to enable this rule.

## ðŸ’¡ Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad">

```js
/*eslint es-x/no-function-prototype-bind: [error, { aggressive: true }] */
foo.bind(this);

var foo = (function() {
    return this.bar
}).bind(this)
```

</eslint-playground>

## ðŸ”§ Options

This rule has an option.

```yaml
rules:
  es-x/no-function-prototype-bind: [error, { aggressive: false }]
```

### aggressive: boolean

Configure the aggressive mode for only this rule.
This is prior to the `settings['es-x'].aggressive` setting.

## ðŸš€ Version

This rule was introduced in v5.0.0.

## ðŸ“š References

- [Rule source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/lib/rules/no-function-prototype-bind.js)
- [Test source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/tests/lib/rules/no-function-prototype-bind.js)
