---
title: "es-x/no-iterator-prototype-filter"
description: "disallow the `Iterator.prototype.filter` method"
---

# es-x/no-iterator-prototype-filter
> disallow the `Iterator.prototype.filter` method

- â— <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
- âœ… The following configurations enable this rule: [no-iterator-helpers] and [no-new-in-esnext]

This rule reports ES2025 [`Iterator.prototype.filter`](https://github.com/tc39/proposal-iterator-helpers) as errors.\

## ğŸ’¡ Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad">

```js
/*eslint es-x/no-iterator-prototype-filter: error */
const result = naturals()
  .filter(n => n % 2);

function* naturals() {
  // ...
}
```

</eslint-playground>

## ğŸ“š References

- [Rule source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/lib/rules/no-iterator-prototype-filter.js)
- [Test source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/tests/lib/rules/no-iterator-prototype-filter.js)

[no-iterator-helpers]: ../configs/index.md#no-iterator-helpers
[no-new-in-esnext]: ../configs/index.md#no-new-in-esnext
