---
title: "es-x/no-intl-supportedvaluesof"
description: "disallow the `Intl.supportedValuesOf` method"
since: "v6.0.0"
---

# es-x/no-intl-supportedvaluesof
> disallow the `Intl.supportedValuesOf` method

- âœ… The following configurations enable this rule: [no-new-in-es2022-intl-api], [restrict-to-es-intl-api-1st-edition], [restrict-to-es2015-intl-api], [restrict-to-es2016-intl-api], [restrict-to-es2017-intl-api], [restrict-to-es2018-intl-api], [restrict-to-es2019-intl-api], [restrict-to-es2020-intl-api], and [restrict-to-es2021-intl-api]

This rule reports  ES2022 Intl API `Intl.supportedValuesOf` method as errors.

## ðŸ’¡ Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad">

```js
/*eslint es-x/no-intl-supportedvaluesof: error */
const collation = Intl.supportedValuesOf("collation")
```

</eslint-playground>

## ðŸš€ Version

This rule was introduced in v6.0.0.

## ðŸ“š References

- [Rule source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/lib/rules/no-intl-supportedvaluesof.js)
- [Test source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/tests/lib/rules/no-intl-supportedvaluesof.js)

[no-new-in-es2022-intl-api]: ../configs/index.md#no-new-in-es2022-intl-api
[restrict-to-es-intl-api-1st-edition]: ../configs/index.md#restrict-to-es-intl-api-1st-edition
[restrict-to-es2015-intl-api]: ../configs/index.md#restrict-to-es2015-intl-api
[restrict-to-es2016-intl-api]: ../configs/index.md#restrict-to-es2016-intl-api
[restrict-to-es2017-intl-api]: ../configs/index.md#restrict-to-es2017-intl-api
[restrict-to-es2018-intl-api]: ../configs/index.md#restrict-to-es2018-intl-api
[restrict-to-es2019-intl-api]: ../configs/index.md#restrict-to-es2019-intl-api
[restrict-to-es2020-intl-api]: ../configs/index.md#restrict-to-es2020-intl-api
[restrict-to-es2021-intl-api]: ../configs/index.md#restrict-to-es2021-intl-api
