# disallow template literals with invalid escape sequences (es/no-malformed-template-literals)

This rule reports ES2018 [template literals with invalid escape sequences](https://github.com/tc39/proposal-template-literal-revision#readme) as errors.

## Examples

â›” Examples of **incorrect** code for this rule:

```js
tag`\unicode`
```

## ðŸ“š References

- [Rule source](../../lib/rules/no-malformed-template-literals.js)
- [Test source](../../tests/lib/rules/no-malformed-template-literals.js)
