# disallow the `Math.trunc` method (es/no-math-trunc)

This rule reports ES2015 `Math.trunc` method as errors.

## Examples

â›” Examples of **incorrect** code for this rule:

```js
const n = Math.trunc(value)
```

## ðŸ“š References

- [Rule source](https://github.com/mysticatea/eslint-plugin-es/blob/v1.2.0/lib/rules/no-math-trunc.js)
- [Test source](https://github.com/mysticatea/eslint-plugin-es/blob/v1.2.0/tests/lib/rules/no-math-trunc.js)
