---
title: "es-x/no-nonstandard-error-properties"
description: "disallow non-standard static properties on `Error` class"
---

# es-x/no-nonstandard-error-properties
> disallow non-standard static properties on `Error` class

- â— <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>

This rule reports non-standard static properties on `Error` class as errors.

## ğŸ’¡ Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad">

```js
/*eslint es-x/no-nonstandard-error-properties: error */
Error.unknown();
```

</eslint-playground>

## ğŸ“š References

- [Rule source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/lib/rules/no-nonstandard-error-properties.js)
- [Test source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/tests/lib/rules/no-nonstandard-error-properties.js)
