---
title: "es-x/no-number-issafeinteger"
description: "disallow the `Number.isSafeInteger` method"
since: "[eslint-plugin-es] v1.2.0"
---

# es-x/no-number-issafeinteger
> disallow the `Number.isSafeInteger` method

- âœ… The following configurations enable this rule: `plugin:es-x/no-new-in-es2015`, `plugin:es-x/restrict-to-es3`, and `plugin:es-x/restrict-to-es5`

This rule reports ES2015 `Number.isSafeInteger` method as errors.

## ğŸ’¡ Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad">

```js
/*eslint es-x/no-number-issafeinteger: error */
const b = Number.isSafeInteger(value)
```

</eslint-playground>

## ğŸš€ Version

This rule was introduced in [eslint-plugin-es] v1.2.0.

[eslint-plugin-es]: https://github.com/mysticatea/eslint-plugin-es

## ğŸ“š References

- [Rule source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/lib/rules/no-number-issafeinteger.js)
- [Test source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/tests/lib/rules/no-number-issafeinteger.js)
