---
title: "es-x/no-nonstandard-string-properties"
description: "disallow non-standard String object properties"
---

# es-x/no-nonstandard-string-properties
> disallow non-standard String object properties

- â— <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>

This rule reports non-standard String object properties as errors.

## ğŸ’¡ Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad">

```js
/*eslint es-x/no-nonstandard-string-properties: error */
String.unknown();
```

</eslint-playground>

## ğŸ”§ Options

This rule has an option.

```jsonc
{
  "rules": {
    "es-x/no-nonstandard-string-properties": [
      "error",
      {
        "allow": []
      }
    ]
  }
}
```

### allow: string[]

An array of non-standard property names to allow.

## ğŸ“š References

- [Rule source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/lib/rules/no-nonstandard-string-properties.js)
- [Test source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/tests/lib/rules/no-nonstandard-string-properties.js)
