# disallow the `Object.defineProperty` method (es/no-object-defineproperty)

This rule reports ES5 `Object.defineProperty` method as errors.

## Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad" code="/*eslint es/no-object-defineproperty: error */
Object.defineProperty(obj, "prop", {})
" />

## ðŸ“š References

- [Rule source](https://github.com/mysticatea/eslint-plugin-es/blob/v2.0.0/lib/rules/no-object-defineproperty.js)
- [Test source](https://github.com/mysticatea/eslint-plugin-es/blob/v2.0.0/tests/lib/rules/no-object-defineproperty.js)
