---
title: "es-x/no-object-seal"
description: "disallow the `Object.seal` method"
since: "[eslint-plugin-es] v3.0.0"
---

# es-x/no-object-seal
> disallow the `Object.seal` method

- âœ… The following configurations enable this rule: `plugin:es-x/no-new-in-es5` and `plugin:es-x/restrict-to-es3`

This rule reports ES5 `Object.seal` method as errors.

## ðŸ’¡ Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad">

```js
/*eslint es-x/no-object-seal: error */
Object.seal(obj)
```

</eslint-playground>

## ðŸš€ Version

This rule was introduced in [eslint-plugin-es] v3.0.0.

[eslint-plugin-es]: https://github.com/mysticatea/eslint-plugin-es

## ðŸ“š References

- [Rule source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/lib/rules/no-object-seal.js)
- [Test source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/tests/lib/rules/no-object-seal.js)
