# es-x/no-object-super-properties
> disallow `super` property accesses in object literals

- âœ… The following configurations enable this rule: `plugin:es-x/no-new-in-es2015`, `plugin:es-x/restrict-to-es3`, and `plugin:es-x/restrict-to-es5`

This rule reports ES2015 `super` property accesses in object literals as errors.

## Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad" code="/*eslint es-x/no-object-super-properties: error */
let a = {
    __proto__: obj,
    f1() { super.a },
    f2() { super.f() }
}
" />

## ðŸ“š References

- [Rule source](https://github.com/ota-meshi/eslint-plugin-es-x/blob/v4.1.0/lib/rules/no-object-super-properties.js)
- [Test source](https://github.com/ota-meshi/eslint-plugin-es-x/blob/v4.1.0/tests/lib/rules/no-object-super-properties.js)
