---
title: "es-x/no-regexp-unicode-property-escapes-2022"
description: "disallow the new values of RegExp Unicode property escape sequences in ES2022"
since: "v6.0.0"
---

# es-x/no-regexp-unicode-property-escapes-2022
> disallow the new values of RegExp Unicode property escape sequences in ES2022

- âœ… The following configurations enable this rule: [no-new-in-es2022], [restrict-to-es3], [restrict-to-es5], [restrict-to-es2015], [restrict-to-es2016], [restrict-to-es2017], [restrict-to-es2018], [restrict-to-es2019], [restrict-to-es2020], and [restrict-to-es2021]

This rule reports the new values of ES2018 [RegExp Unicode property escape sequences](https://github.com/tc39/proposal-regexp-unicode-property-escapes#readme) which were added in ES2022.

For example, the following patterns are valid in ES2022, but syntax error in ES2021 environments:

- `\p{Script=Cpmn}`
- `\p{Script=Cypro_Minoan}`
- `\p{Script=Old_Uyghur}`
- `\p{Script=Ougr}`
- `\p{Script=Tangsa}`
- `\p{Script=Tnsa}`
- `\p{Script=Toto}`
- `\p{Script=Vith}`
- `\p{Script=Vithkuqi}`

## ðŸ’¡ Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad">

```js
/*eslint es-x/no-regexp-unicode-property-escapes-2022: error */
const r1 = /\p{Script=Cpmn}/u
const r2 = /\p{Script=Cypro_Minoan}/u
```

</eslint-playground>

## ðŸš€ Version

This rule was introduced in v6.0.0.

## ðŸ“š References

- [Rule source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/lib/rules/no-regexp-unicode-property-escapes-2022.js)
- [Test source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/tests/lib/rules/no-regexp-unicode-property-escapes-2022.js)

[no-new-in-es2022]: ../configs/index.md#no-new-in-es2022
[restrict-to-es3]: ../configs/index.md#restrict-to-es3
[restrict-to-es5]: ../configs/index.md#restrict-to-es5
[restrict-to-es2015]: ../configs/index.md#restrict-to-es2015
[restrict-to-es2016]: ../configs/index.md#restrict-to-es2016
[restrict-to-es2017]: ../configs/index.md#restrict-to-es2017
[restrict-to-es2018]: ../configs/index.md#restrict-to-es2018
[restrict-to-es2019]: ../configs/index.md#restrict-to-es2019
[restrict-to-es2020]: ../configs/index.md#restrict-to-es2020
[restrict-to-es2021]: ../configs/index.md#restrict-to-es2021
