---
title: "es-x/no-set-prototype-intersection"
description: "disallow the `Set.prototype.intersection` method"
---

# es-x/no-set-prototype-intersection
> disallow the `Set.prototype.intersection` method

- â— <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
- âœ… The following configurations enable this rule: [no-new-in-esnext]

This rule reports ES2025 [`Set.prototype.intersection`](https://github.com/tc39/proposal-set-methods) methods as errors.

This rule is silent by default because it's hard to know types. You need to configure [the aggressive mode](../#the-aggressive-mode) or TypeScript in order to enable this rule.

## ğŸ’¡ Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad">

```js
/*eslint es-x/no-set-prototype-intersection: error */
const a = new Set()
const b = new Set()
a.intersection(b)
```

</eslint-playground>

## ğŸ“š References

- [Rule source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/lib/rules/no-set-prototype-intersection.js)
- [Test source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/tests/lib/rules/no-set-prototype-intersection.js)

[no-new-in-esnext]: ../configs/index.md#no-new-in-esnext
