---
title: "es-x/no-subclassing-builtins"
description: "disallow the subclassing of the built-in classes"
since: "[eslint-plugin-es] v1.2.0"
---

# es-x/no-subclassing-builtins
> disallow the subclassing of the built-in classes

- âœ… The following configurations enable this rule: [no-new-in-es2015], [restrict-to-es3], and [restrict-to-es5]

This rule reports ES2015 subclassing of built-in classes as errors.

The built-in classes include the following classes (constructors):

- `Array`
- `Boolean`
- `Error`
- `RegExp`
- `Function`
- `Map`
- `Number`
- `Promise`
- `Set`
- `String`

## ðŸ’¡ Examples

â›” Examples of **incorrect** code for this rule:

<eslint-playground type="bad">

```js
/*eslint es-x/no-subclassing-builtins: error */
class MyArray extends Array {
    // ...
}
```

</eslint-playground>

## ðŸš€ Version

This rule was introduced in [eslint-plugin-es] v1.2.0.

[eslint-plugin-es]: https://github.com/mysticatea/eslint-plugin-es

## ðŸ“š References

- [Rule source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/lib/rules/no-subclassing-builtins.js)
- [Test source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/tests/lib/rules/no-subclassing-builtins.js)

[no-new-in-es2015]: ../configs/index.md#no-new-in-es2015
[restrict-to-es3]: ../configs/index.md#restrict-to-es3
[restrict-to-es5]: ../configs/index.md#restrict-to-es5
