---
title: "es-x/no-string-prototype-iswellformed-towellformed"
description: "disallow the `String.prototype.{isWellFormed,toWellFormed}` methods"
---

# es-x/no-string-prototype-iswellformed-towellformed
> disallow the `String.prototype.{isWellFormed,toWellFormed}` methods

- ‚ùó <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
- ‚úÖ The following configurations enable this rule: `plugin:es-x/no-new-in-esnext`

## üí° Examples

‚õî Examples of **incorrect** code for this rule:

<eslint-playground type="bad">

```js
/*eslint es-x/no-string-prototype-iswellformed-towellformed: [error, { aggressive: true }] */
"str".isWellFormed()
"str".toWellFormed()
```

</eslint-playground>a

## üîß Options

This rule has an option.

```yaml
rules:
  es-x/no-string-prototype-iswellformed-towellformed: [error, { aggressive: false }]
```

### aggressive: boolean

Configure the aggressive mode for only this rule.
This is prior to the `settings['es-x'].aggressive` setting.

## üìö References

- [Rule source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/lib/rules/no-string-prototype-iswellformed-towellformed.js)
- [Test source](https://github.com/eslint-community/eslint-plugin-es-x/blob/master/tests/lib/rules/no-string-prototype-iswellformed-towellformed.js)
